src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -1213,6 +1211,192 @@ private static float MinMaxCore<TMinMaxOperator>(ReadOnlySpan<float> x)
             }
         }
 
+        /// <remarks>
+        /// This is the same as <see cref="Aggregate{TLoad, TAggregate}(ReadOnlySpan{float})"/>,
+        /// except it early exits on NaN.
+        /// </remarks>
+        private static int IndexOfMinMaxCore<TIndexOfMinMax>(ReadOnlySpan<float> x) where TIndexOfMinMax : struct, IIndexOfOperator
+        {
+            // This matches the IEEE 754:2019 `maximum`/`minimum` functions.
+            // It propagates NaN inputs back to the caller and
+            // otherwise returns the greater of the inputs.
+            // It treats +0 as greater than -0 as per the specification.
+
+#if NET8_0_OR_GREATER
+            if (Vector512.IsHardwareAccelerated && x.Length >= Vector512<float>.Count)
+            {
+                ref float xRef = ref MemoryMarshal.GetReference(x);
+                Vector512<int> resultIndex = Vector512.Create(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
+                Vector512<int> curIndex = Vector512.Create(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
+                Vector512<int> increment = Vector512.Create(Vector512<float>.Count);
+
+                // Load the first vector as the initial set of results, and bail immediately
+                // to scalar handling if it contains any NaNs (which don't compare equally to themselves).
+                Vector512<float> result = Vector512.LoadUnsafe(ref xRef, 0), current;
+                if (!Vector512.EqualsAll(result, result))
```

Comment:
Operators work and the comment helps clarify intent of the check:
```suggestion
                if (result != result) // check if any are NaN
```

Reply:
Alternatively, this would probably be more efficient as some form of:
```csharp
Vector512<float> nanMask = Vector512.Equals(result, result);

if (nanMask != Vector512<float>.Zero)
{
    return i + IndexOfFirstMatch(nanMask);
}
```

Reply:
Where `IndexOfFirstMatch` is functionally:
```csharp
static int IndexOfFirstMatch(Vector512<float> mask)
{
    return BitOperations.TrailingZeroCount(mask.ExtractMostSignificantBits);
}
```

Reply:
ALright, I have converted it to this pattern.